cpp/src/parquet/types.cc

```diff
@@ -1619,6 +1658,209 @@ class LogicalType::Impl::Float16 final : public LogicalType::Impl::Incompatible,
 
 GENERATE_MAKE(Float16)
 
+namespace {
+void WriteCrsKeyAndValue(const std::string& crs, std::ostream& json) {
+  // There is no restriction on the crs value here, and it may contain quotes
+  // or backslashes that would result in invalid JSON if unescaped.
+  json << R"(, "crs": ")";
+  for (char c : crs) {
+    switch (c) {
+      case '"':
+        json << R"(\")";
+        break;
+      case '\\':
+        json << R"(\\)";
+        break;
+      default:
```

Comment:
Characters below 32 should also be encoded as unicode escapes (`\uXXXX`), I believe.